ibis/backends/sql/compilers/duckdb.py

```diff
@@ -177,16 +177,19 @@ def visit_ArrayRepeat(self, op, *, arg, times):
         return self.f.flatten(self.f.list_apply(self.f.range(times), func))
 
     def visit_ArraySlice(self, op, *, arg, start, stop):
+        import duckdb
```

Comment:
The compilers **must** be usable without any dependencies installed, so you can't do this.

Reply:
Ah, didn't know this. Thanks!